source/extensions/utility.hpp

```diff
@@ -85,4 +87,39 @@ static constexpr void IterateFunction(auto&& functor) {
     }
 }
 
+//! Simple, thread safe singleton pattern. Instance created on first call to `GetSingleton()`.
```

Comment:
Not thread safe anymore
```suggestion
//! Simple (not thread safe) singleton class. Instance created on first call to `GetSingleton()`.
```